tests/testthat/test-forward_select.R

```diff
@@ -0,0 +1,70 @@
+#
+# Tests for forward_selection()
+#
+
+#' Sample custom scorer that fits a model and returns
+#' an appropriate score for the feature selection problem.
+#'
+#' @param data A data.frame with features and response columns.
+#'
+#' @return The mean squared error (MSE) of the model
+scorer <- function(data) {
+   model <- lm(Y ~ ., data)
+   return(mean(model$residuals^2))
+}
+
+# This test creates a dataset that has 5 features that are are used to compute `Y`.
+# The remaining features are independent of `Y`.
+# This test should select the 5 feature columns used to compute `Y`.
+testthat::test_that("relevant features remain", {
+  # Create dataframe and remove Y true column; keep Y.
+  set.seed(0)
+  data <- dplyr::select(tgp::friedman.1.data(), -Ytrue)
+  X <- data[1:(length(data)-1)]
+  y <- data[length(data)]
+
+  # Run Test compare scores with expected ones
+  set.seed(1230)
+  results <- forward_select(scorer, X, y, 4, 4)
+  testthat::expect_setequal(results, c(1, 2, 4, 5))
+
+  # X and y are Dataframes
```

Comment:
A lot of the test cases below should be separate and outside of the relevant feature test.